### PR TITLE
Using Red Hat Universal Base Image 8 as the new Strimzi base image

### PR DESCRIPTION
**Pull request by scholzj**
This proposal suggests to use UBI8 as the new base image for Strimzi projects.

### CODE REVIEW

**Review by ppatierno — APPROVED** [review]

File: 023-using-ubi8-as-base-image.md [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+
+## Current situation
+
+Strimzi container images are currently based on the CentOS 7 container images.
+The CentOS 7 container images are stored on Docker Hub and used to build all Strimzi images.
+
+## Motivation
+
+Using the CentOS 7 image has two main advantages:
+* It is hosted on Docker Hub whihc has limited number of anonymous pulls (see [proposal 14](https://github.com/strimzi/proposals/blob/main/014-move-docker-images-to-quay.io.md) for more details).
```
Comment (Frawless) [inline 023-using-ubi8-as-base-image.md:13]:
```suggestion
* It is hosted on Docker Hub which has limited number of anonymous pulls (see [proposal 14](https://github.com/strimzi/proposals/blob/main/014-move-docker-images-to-quay.io.md) for more details).
```

File: 023-using-ubi8-as-base-image.md [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+* CentOS 7 images are not released too often (at the time of writing, the image is 2 months old), so we need to install lot of updates while building the images
+* CentOS 7 received CVE fixes in batches often onyl after they are available in other images
```
Comment (Frawless) [inline 023-using-ubi8-as-base-image.md:15]:
```suggestion
* CentOS 7 received CVE fixes in batches often only after they are available in other images
```

File: 023-using-ubi8-as-base-image.md [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+This proposal suggests to use Red Hat UBI8 as the new Strimzi base image.
+
+## Current situation
+
+Strimzi container images are currently based on the CentOS 7 container images.
+The CentOS 7 container images are stored on Docker Hub and used to build all Strimzi images.
+
+## Motivation
+
+Using the CentOS 7 image has two main advantages:
```
Comment (Frawless) [inline 023-using-ubi8-as-base-image.md:12]:
```suggestion
Using the CentOS 7 image has three main disadvantages:
```

Comment (Frawless) [reply 023-using-ubi8-as-base-image.md:12]: Not sure if I understand correctly, but it makes better sense to me

File: 023-using-ubi8-as-base-image.md [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+
+If approved, the Dockerfiles in the affected Strimzi projects will be updated to use the `ubi-minimal` image.
+
+## Compatibility
+
+There are no compatibility issues.
+
+## Rejected alternatives
+
+For this proposal, I considered mainly the Re dHat and CentOS images.
```
Comment (Frawless) [inline 023-using-ubi8-as-base-image.md:55]:
```suggestion
For this proposal, I considered mainly the RedHat and CentOS images.
```

**Review by Frawless — APPROVED** [review]
Just some nits, I agree with this.

File: 023-using-ubi8-as-base-image.md [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+But to my knowledge, they affect only the CentOS 8 version.
+CentOS Streams and CentOS 7 remain unchanged.
+CentOS 7 should receive maintenance updates until the year 2024.
+So there is no time pressure to move to other base image caused by these changes._
+
+## Proposal
+
+Strimzi should move to use Red Hat Universal Base Image 8 (UBI 8) as a base image.
+UBI8 is based on Red Hat Enterprise Linux 8 and is available to everyone for free (it does not include any Red Hat support).
+It can be pulled without any registration from `registry.access.redhat.com/ubi8/ubi-minimal:latest` and be redistributed (i.e. we can push images build on this base image into our repositories).
```
Comment (tombentley) [inline 023-using-ubi8-as-base-image.md:27]: Is there some official license of statement about exactly what it and isn't allowed to be done with these images. For example, that might restrict people who are using custom images (I can't imagine that there are such restrictions, but it's impossible to be sure without seeing some official statement). Can we add it to the proposal, just for the record?

**Review by sknot-rh — APPROVED** [review]

File: 023-using-ubi8-as-base-image.md (hunk 5 above)
Comment (scholzj) [reply 023-using-ubi8-as-base-image.md:27]: I added direct link to the license as well as the related FAQ.

**Comment by scholzj** [comment]
@tombentley Is this ok with you now after the last changes?

**Review by tombentley — APPROVED** [review]

**Comment by scholzj** [comment]
Approved.

**Pull request merged**